From 56a318850eadf33c586d84e676bb0e98040e7703 Mon Sep 17 00:00:00 2001
From: Arulanbu Balusamy <quic_abalusam@quicinc.com>
Date: Sun, 5 May 2024 13:15:55 +0530
Subject: [PATCH] wifi: mac80211: Add the MLD vif support to handle beacon
 loss.


In ieee80211_beacon_loss function triggers the QoS null
packet for some time from STA to check whether the AP is
alive. But there is no support for Multi Link VAP's on
to send the Null packet. So remove the multi link condition
and allow the STA to send the Qos Null packets even the vif
is MLD.

Fixes: Ic102d5a6ce6 ("ath12k: Add support for STA mode to
trigger the disconnection by using ieee80211_beacon_loss.")


Signed-off-by: Arulanbu Balusamy <quic_abalusam@quicinc.com>
---
 net/mac80211/mlme.c | 8 +-------
 1 file changed, 1 insertion(+), 7 deletions(-)

--- a/net/mac80211/mlme.c
+++ b/net/mac80211/mlme.c
@@ -3220,9 +3220,6 @@ static void ieee80211_mgd_probe_ap_send(
 	u8 unicast_limit = max(1, max_probe_tries - 3);
 	struct sta_info *sta;
 
-	if (WARN_ON(ieee80211_vif_is_mld(&sdata->vif)))
-		return;
-
 	/*
 	 * Try sending broadcast probe requests for the last three
 	 * probe requests after the first ones failed since some
@@ -3268,9 +3265,6 @@ static void ieee80211_mgd_probe_ap(struc
 	struct ieee80211_if_managed *ifmgd = &sdata->u.mgd;
 	bool already = false;
 
-	if (WARN_ON_ONCE(ieee80211_vif_is_mld(&sdata->vif)))
-		return;
-
 	if (!ieee80211_sdata_running(sdata))
 		return;
 
